typescript/cli/src/registry/MergedRegistry.ts

```diff
@@ -0,0 +1,154 @@
+import { Logger } from 'pino';
+
+import {
+  BaseRegistry,
+  ChainAddresses,
+  GithubRegistry,
+  IRegistry,
+  RegistryContent,
+  RegistryType,
+} from '@hyperlane-xyz/registry';
+import { LocalRegistry } from '@hyperlane-xyz/registry/local';
+import {
+  ChainMap,
+  ChainMetadata,
+  ChainName,
+  WarpCoreConfig,
+} from '@hyperlane-xyz/sdk';
+import {
+  isHttpsUrl,
+  objKeys,
+  objMerge,
+  rootLogger,
+} from '@hyperlane-xyz/utils';
+
+export interface MergedRegistryOptions {
+  registryUris: Array<string>;
+  isDryRun?: boolean;
```

Comment:
surprised to see this here?

Reply:
I need it to have this registry behave differently when writing during a dry run, which is still TODO.
We can't simply point to the local registry to a diff root path because we may still need to read the data there